infra/gcp/istio-prow-private/cluster.tf

```diff
@@ -0,0 +1,75 @@
+module "prow" {
+  source       = "../modules/gke-cluster"
+  cluster_name = "prow"
+  project_name = local.project_id
+  # ARM node pools are (currently) only available here, so ensure we use this region.
+  cluster_location = "us-central1-f"
+}
+module "prow_node" {
+  source = "../modules/gke-nodepool"
+  name   = "test"
+
+  project_name = local.project_id
+  cluster_name = module.prow.cluster.name
+  location     = module.prow.cluster.location
+
+  machine_type  = "e2-standard-16"
+  initial_count = 1
+  min_count     = 1
+  max_count     = 15
+
+  disk_size_gb = 256
+  disk_type    = "pd-ssd"
+
+  service_account = module.prow.cluster_node_sa.email
+
+  labels = {
+    "testing" = "test-pool"
+  }
+}
+module "prow_node_build" {
+  source = "../modules/gke-nodepool"
+  name   = "build"
+
+  project_name = local.project_id
+  cluster_name = module.prow.cluster.name
+  location     = module.prow.cluster.location
+
+  machine_type  = "n1-highmem-64"
+  initial_count = 0
+  min_count     = 0
+  max_count     = 10
+
+  disk_size_gb = 2000
+  disk_type    = "pd-ssd"
+
+  service_account = module.prow.cluster_node_sa.email
+
+  labels = {
+    "testing" = "build-pool"
+  }
+}
+module "prow_node_arm" {
+  source = "../modules/gke-nodepool"
+  name   = "arm"
+
+  project_name = local.project_id
+  cluster_name = module.prow.cluster.name
+  location     = module.prow.cluster.location
+
+  machine_type  = "t2a-standard-16"
+  initial_count = 2
+  max_count     = 60
+  min_count     = 2
+
+  disk_size_gb = 256
+  disk_type    = "pd-ssd"
+
+  service_account = module.prow.cluster_node_sa.email
+
+  spot = true
```

Comment:
why spot instances here but not on the build-pool?

Reply:
will put a comment in the code, but - GCP is only allowing non-trivial quotas for t2a nodes using spot instances